### PR TITLE
[video_player] Exposes VideoPlayerWebOptions.

### PR DESCRIPTION
**Pull request by ditman**
Exports types `VideoPlayerWebOptions` and `VideoPlayerWebOptionsControls` to customize the `webOptions` field in `VideoPlayerOptions` objects.

Forwards `webOptions` to the web implementation.

## Usage

Can be used as follows:

```dart
_controller = VideoPlayerController.asset(
  'assets/Butterfly-209.mp4',
  videoPlayerOptions: const VideoPlayerOptions(
    webOptions: VideoPlayerWebOptions(
        controls: VideoPlayerWebOptionsControls.enabled(
          allowDownload: false,
          allowFullscreen: false,
          allowPlaybackRate: false,
          allowPictureInPicture: false,
        ),
        allowContextMenu: false,
        allowRemotePlayback: false,
     ),
  ),
);
```

Bildschirmfoto 2023-07-06 um 17 04 59

## Issues

* Fixes: https://github.com/flutter/flutter/issues/150327
* Fixes: https://github.com/flutter/flutter/issues/64397
* Fixes: https://github.com/flutter/flutter/issues/62192
* Fixes part of: https://github.com/flutter/flutter/issues/88151
* Lands: https://github.com/flutter/packages/pull/3259 (adds missing test)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

Co-authored-by: James Leahy <defuncart@gmail.com>

### CODE REVIEW

File: packages/video_player/video_player/pubspec.yaml [hunk 1]
```diff
@@ -27,7 +27,7 @@ dependencies:
   html: ^0.15.0
   video_player_android: ^2.3.5
   video_player_avfoundation: ^2.5.6
-  video_player_platform_interface: ">=6.1.0 <7.0.0"
+  video_player_platform_interface: ">=6.2.0 <7.0.0"
```
Comment (stuartmorgan) [inline packages/video_player/video_player/pubspec.yaml:30]: This can just be `^6.2.0`.

**Review by stuartmorgan — APPROVED** [review]
LGTM.

We really need to rework the way platform options work, since having to hard-code platform-specific logic in the app-facing package is gross, but since it was already mostly landed this way that's a problem for a future redesign.

File: packages/video_player/video_player/pubspec.yaml (hunk 1 above)
Comment (ditman) [reply packages/video_player/video_player/pubspec.yaml:30]: You're totally right, let me fix this so we don't propagate the ">=<" syntax further than strictly needed.

**Comment by ditman** [comment]
> We really need to rework the way platform options work, since having to hard-code platform-specific logic in the app-facing package is gross, but since it was already mostly landed this way that's a problem for a future redesign.

Yes, it's clear that platform-specific code ends popping up in the plugin API :/

**Pull request merged**